lapis/src/main/kotlin/org/genspectrum/lapis/model/AdvancedQueryCustomListener.kt

```diff
@@ -0,0 +1,534 @@
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.IntTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+        }
+    }
+
+    override fun enterRegexMetadataQuery(ctx: AdvancedQueryParser.RegexMetadataQueryContext) {
+        val metadataName = ctx.geneOrName().text
+        val metadataValue = ctx.value().text.trim('\'')
+
+        val field: SequenceFilterField? = allowedSequenceFilterFields.fields[metadataName.lowercase(Locale.US)]
+        field ?: throw BadRequestException("Metadata field $metadataName does not exist", null)
+        when (field.type) {
+            SequenceFilterFieldType.String -> {
+                expressionStack.addLast(StringEquals(field.name, metadataValue))
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.Lineage -> {
+                val lineage = mapToLineageFilter(
+                    field.name,
+                    metadataValue,
+                )
+                expressionStack.addLast(lineage)
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+
+            else -> {
+                throw BadRequestException(
+                    "Expression contains symbols not allowed for metadata field of type ${field.type} (allowed symbols: a-z, A-Z, 0-9, ., *, -, _)",
+                    null,
+                )
+            }
+        }
+    }
+
+    override fun enterIsNullQuery(ctx: AdvancedQueryParser.IsNullQueryContext) {
+        val metadataName = ctx.geneOrName().text
+
+        val field: SequenceFilterField? = allowedSequenceFilterFields.fields[metadataName.lowercase(Locale.US)]
+        field ?: throw BadRequestException("Metadata field $metadataName does not exist", null)
```

Comment:
I think this would also benefit from using the metadata fields instead of the `SequenceFilterField`s: We could simply check whether the `metadataName` is in the config. If yes, continue. If no, `throw "$metadataName is not a field in the config"`.